### ISSUE TITLE
Explicitly requested binary encoding is not respected when binding parameters (v 4.0.4)

### ISSUE THREAD

**Issue by karol-pilch** [Azure/msphpsql#90]
I'm using latest to date driver that reports as 4.0.8429.1 in `php_info()`.

The following code:

```php
$pdo = new PDO('my connection string', 'username', 'password');

$value = 'Some string value.';
$name = 'name';

$sql = "UPDATE my_table SET [value] = :value WHERE [name] = :name";

$stmt = $pdo->prepare($sql);

$stmt->bindParam(':value', $value, PDO::PARAM_LOB, 0, PDO::SQLSRV_ENCODING_BINARY);
$stmt->bindParam(':name', $name);

$stmt->execute();
$stmt->closeCursor();
```

Produces the following SQL when using version for PHP 5.6 (it doesn't show version number in `php_info()`):

```sql
declare @p1 int
set @p1=1
exec sp_prepexec @p1 output,N'@P1 varbinary(18),@P2 nvarchar(4)',N'UPDATE my_table SET [value] = @P1 WHERE [name] = @P2',0x536F6D6520737472696E672076616C75652E,N'name'
select @p1
```

which works fine. However, the latest version produces the following SQL:

```sql
declare @p1 int
set @p1=NULL
exec sp_prepexec @p1 output,N'@P1 nvarchar(18),@P2 nvarchar(4)',N'UPDATE my_table SET [value] = @P1 WHERE [name] = @P2',N'Some string value.',N'name'
select @p1
```

The execution of the latter fails with the error "Implicit conversion from data type nvarchar to varbinary(max) is not allowed. Use the CONVERT function to run this query.".

This example shows that explicitly passing `PDO::SQLSRV_ENCODING_BINARY` to `bindParam` method does not trigger the conversion to binary.

I'm happy to provide more details if necessary.

Thanks!

**Comment by csucla2015** [comment]
Hello @karol-pilch 
What version of PHP are you using? We did some investigation and this issue is similar to https://github.com/Azure/msphpsql/issues/83.
The issue with bindParam should be fixed in PHP 7.0.6

Thanks,
Meet

**Comment by meet-bhagdev** [comment]
Hello @karol-pilch 
What version of PHP are you using? We did some investigation and this issue is similar to https://github.com/Azure/msphpsql/issues/83
The issue with bindParam should be fixed in PHP 7.0.6

Thanks,
Meet

**Comment by karol-pilch** [comment]
Hi @meet-bhagdev, thanks for the prompt reply. The issue seems to be fixed in 7.0.6 - sorry for not checking that before reporting.

Best,

Karol
